pkg/envtest/komega/komega.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package komega
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// komega has Gomega Matchers that use the controller-runtime client.
+type komega struct {
+	ctx          context.Context
+	g            gomega.Gomega
+	client       client.Client
+	extras       []interface{}
+	timeout      time.Duration
+	pollInterval time.Duration
+}
+
+var _ Komega = &komega{}
+
+// NewKomega creates a new instance with a given client.
+func NewKomega(c client.Client) *komega {
+	return &komega{
+		g:      gomega.Default,
+		client: c,
+	}
+}
+
+// NewKomegaWithT creates a new instance with a given client and a Gomega derived from a given testing.T.
+func NewKomegaWithT(t *testing.T, c client.Client) *komega {
+	return &komega{
+		g:      gomega.NewWithT(t),
+		client: c,
+	}
+}
+
+// NewKomegaWithG creates a new instance with a given client and gomega.
+func NewKomegaWithG(g gomega.Gomega, c client.Client) *komega {
+	return &komega{
+		g:      g,
+		client: c,
+	}
+}
+
+// WithContext returns a copy using the given Context.
+func (k komega) WithContext(ctx context.Context) Komega {
+	k.ctx = ctx
+	return &k
+}
+
+// WithExtras returns a copy using the given extras.
+func (k komega) WithExtras(extras ...interface{}) Komega {
+	k.extras = extras
+	return &k
+}
+
+// WithPollInterval returns a copy using the given poll interval.
+func (k komega) WithPollInterval(interval time.Duration) Komega {
+	k.pollInterval = interval
+	return &k
+}
+
+// WithTimeout returns a copy using the given timeout.
+func (k komega) WithTimeout(timeout time.Duration) Komega {
+	k.timeout = timeout
+	return &k
+}
+
+// context returns the matcher context if one has been set.
+// Else it returns the context.TODO().
+func (k *komega) context() context.Context {
+	if k.ctx == nil {
+		return context.TODO()
```

Comment:
Should this be `context.Background()`? 

```go
// TODO returns a non-nil, empty Context. Code should use context.TODO when
// it's unclear which Context to use or it is not yet available (because the
// surrounding function has not yet been extended to accept a Context
// parameter).
```